api/src/main/java/com/netflix/iceberg/DataOperations.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.netflix.iceberg;
+
+/**
+ * Data operations that produce snapshots.
+ * <p>
+ * A snapshot can return the operation that created the snapshot to help other components ignore
+ * snapshots that are not needed for some tasks. For example, snapshot expiration does not need to
+ * clean up deleted files for appends, which have no deleted files.
+ */
+public class DataOperations {
```

Comment:
Why did you chose to not make this an Enum?

Reply:
Trying to be forward-compatible. Iceberg should recognize known data operations constants, but not reject unknown ones.